.cargo/config_fast_builds.toml

```diff
@@ -16,6 +17,7 @@ rustflags = [
 rustflags = [
   "-Clink-arg=-fuse-ld=/usr/local/opt/llvm/bin/ld64.lld", # Use LLD Linker
   "-Zshare-generics=y",                                   # (Nightly) Make the current crate share its generic instantiations
+  "-Zthreads=8",                                          # (Nightly) Use improved multithreading with 8 threads.
```

Comment:
Can any MacOS users test if this works well? It might reduce performance on MacOS or even not work at all.

Reply:
After testing it on my M1 Mac, this option actually seems to slow down the compilation process. I built with the `--timings` option to generate the following charts. The time increases from 59 seconds to 63.6 seconds.

Before:

<img width="457" alt="image" src="https://github.com/bevyengine/bevy/assets/59022059/5d8017d7-de6c-4612-9394-da989e1011f1">

<img width="1300" alt="image" src="https://github.com/bevyengine/bevy/assets/59022059/08299147-abb5-499e-bce0-59c4f65004e1">

After:

<img width="457" alt="image" src="https://github.com/bevyengine/bevy/assets/59022059/e4f99322-c198-4ab7-b704-694a1fb13e5f">

<img width="1430" alt="image" src="https://github.com/bevyengine/bevy/assets/59022059/fb13879b-4c82-4fc5-998a-d354f89efdd1">

Reply:
Is this an average or just one test?

Reply:
That was just one test. Here's my results running `cargo clean && cargo +nightly build` with and without `-Zthreads=0` commented out.

### Before

- 1:03s
- 1:06s
- 1:11s
- 1:12s
- 1:08s
- 0:59.4s
- 1:03s

Average: 1:06.05s

### After

- 1:05s
- 1:11s
- 1:13s
- 1:20s
- 1:07s
- 1:11s
- 1:12s

Average: 1:11.29s